python/pyspark/sql/connect/functions.py

```diff
@@ -1988,13 +1988,70 @@ def split(str: "ColumnOrName", pattern: str, limit: int = -1) -> Column:
 split.__doc__ = pysparkfuncs.split.__doc__
 
 
+def rlike(string: "ColumnOrName", pattern: Union[str, Column]) -> Column:
```

Comment:
for the python side, let's use `ColumnOrName` augments. that is, when `pattern` is a `str`, treat it as a column name

see discussion here https://github.com/apache/spark/pull/41505#discussion_r1222631192